src/software/logger/protobuf_sink.cpp

```diff
@@ -20,21 +20,30 @@ void ProtobufSink::sendProtobuf(g3::LogMessageMover log_entry)
 
     if (level.value == VISUALIZE.value)
     {
-        std::string msg = log_entry.get().message();
-        size_t pos      = msg.find(TYPE_DELIMITER);
+        std::string msg       = log_entry.get().message();
+        size_t file_name_pos  = msg.find(TYPE_DELIMITER);
+        std::string file_name = msg.substr(0, file_name_pos);
 
-        std::string proto_type_name  = msg.substr(0, pos);
-        std::string serialized_proto = msg.substr(pos + TYPE_DELIMITER.length());
+        size_t proto_type_name_pos = msg.find(TYPE_DELIMITER, file_name_pos + 1);
+        std::string proto_type_name =
+            msg.substr(file_name_pos + TYPE_DELIMITER.length(), proto_type_name_pos - 3);
```

Comment:
instead of the  - 3 can you replace it with `TYPE_DELIMITER.length()`

Reply:
Updated